### PR TITLE
Add Infomaniak provider

### PR DESCRIPTION
**Pull request by L3o-pold**

### CODE REVIEW

File: src/Infomaniak/InfomaniakExtendSocialite.php [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+class InfomaniakExtendSocialite
+{
+    /**
+     * Register the provider.
+     *
+     * @param  \SocialiteProviders\Manager\SocialiteWasCalled  $socialiteWasCalled
+     * @return void
+     */
+    public function handle(SocialiteWasCalled $socialiteWasCalled)
```
Comment (lucasmichot) [inline src/Infomaniak/InfomaniakExtendSocialite.php:15]:
```suggestion
    public function handle(SocialiteWasCalled $socialiteWasCalled): void
```

File: src/Infomaniak/Provider.php [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+use GuzzleHttp\RequestOptions;
+use SocialiteProviders\Manager\OAuth2\AbstractProvider;
+use SocialiteProviders\Manager\OAuth2\User;
+
+class Provider extends AbstractProvider
+{
+    const IDENTIFIER = 'INFOMANIAK';
+
+    /**
+     * {@inheritdoc}
```
Comment (lucasmichot) [inline src/Infomaniak/Provider.php:14]: You can drop this docblock

File: src/Infomaniak/Provider.php [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+     */
+    protected $scopes = ['openid', 'email', 'profile'];
+
+    /**
+     * {@inheritdoc}
```
Comment (lucasmichot) [inline src/Infomaniak/Provider.php:19]: Same here

File: src/Infomaniak/Provider.php [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+     */
+    protected $scopeSeparator = ' ';
+
+    /**
+     * {@inheritdoc}
```
Comment (lucasmichot) [inline src/Infomaniak/Provider.php:24]: Same here

File: src/Infomaniak/Provider.php [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+     */
+    protected function getAuthUrl($state)
+    {
+        return $this->buildAuthUrlFromBase('https://login.infomaniak.com/authorize', $state);
+    }
+
+    /**
+     * {@inheritdoc}
```
Comment (lucasmichot) [inline src/Infomaniak/Provider.php:32]: Same here

File: src/Infomaniak/composer.json [hunk 6]
```diff
@@ -0,0 +1,21 @@
+{
```
Comment (lucasmichot) [inline src/Infomaniak/composer.json:1]: Can you also add the `keywords` and `support` section?

**Review by lucasmichot — CHANGES_REQUESTED** [review]
Thanks for the PR Léopold, some changes requested

**Comment by L3o-pold** [comment]
I used https://github.com/SocialiteProviders/Generators so maybe we needs to change it too.

**Review by lucasmichot — APPROVED** [review]
GTM @atymic

**Comment by lucasmichot** [comment]
@L3o-pold , can you make also a PR on https://github.com/SocialiteProviders/website ?

**Comment by L3o-pold** [comment]
done: https://github.com/SocialiteProviders/website/pull/68

**Review by atymic — APPROVED** [review]

**Pull request merged**